### PR TITLE
switch to using PKG

### PR DESCRIPTION
**Pull request by gdams**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

**Comment by karianna** [comment]
I'll echo this - do folks who use brew expect a PKG?  My guess is probably not...

**Comment by gdams** [comment]
@VonUniGE 

>     1. Why is there no sha256 checksum?
We can add one, I figured it wasn't neccessary as macOS will flag the pkg as corrupt if it isn't codesigned.

>     2. The JVM is not registred system wide: nothing appears with `/usr/libexec/java_home -V`
This is working for me:
```
   1.8.0_202, x86_64:	"AdoptOpenJDK (OpenJ9) 8"	/Library/Java/JavaVirtualMachines/adoptopenjdk-8-openj9.jdk/Contents/Home
```

>     3. What are the advantages of PKG install over the current method?
It heavily simplifies the cask files as all of the symlink and plist stuff happens when the installer is built. Many cask files use PKG files and this also means that we are better aligned with our installers that we ship.

File: Casks/adoptopenjdk8-openj9.rb [hunk 1]
```diff
@@ -1,68 +1,16 @@
 cask 'adoptopenjdk8-openj9' do
   version '8,202:b08'
-  sha256 '9ed76d2e4885bc1c7d9c937abaaf3739bddd2cd7c6deca5a46b79e94d62d421e'
+  sha256 :no_check
```
Comment (karianna) [inline Casks/adoptopenjdk8-openj9.rb]: Add a comment as to why no sha256 check in that case

Comment (gdams) [reply Casks/adoptopenjdk8-openj9.rb]: yeah can add Sha256 if we are happy with everything else?

**Comment by VonUniGE** [comment]
1. Ok, to have the sha256 checksum.
2. Ok, the JVM is correctly registered
3. Ok, to use PKG because of two advantages:
 a. Simplify cask files
 b. Reuse work done in PKG files

**Comment by gdams** [comment]
@VonUniGE updated PTAL

**Pull request merged**